flow/ProtocolVersion.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+#include <cstdint>
+
+namespace fdb_versions {
+
+#define VERSION_FEATURE(x)                                                                                             \
+	constexpr bool has##x() const { return static_cast<const P*>(this)->version() >= protocolVersion; }
+
+template<class P>
+struct v5_5 {
+	static constexpr uint64_t protocolVersion = 0x0FDB00A551000000LL;
+
+	VERSION_FEATURE(MultiVersionClient)
+};
+
+template<class P>
+struct v5_6 : v5_5<P> {
+	static constexpr uint64_t protocolVersion = 0x0FDB00A560010001LL;
+
+	VERSION_FEATURE(TagLocality)
+};
+
+template <class P>
+struct v6_0 : v5_6<P> {
+	static constexpr uint64_t protocolVersion = 0x0FDB00B060000001LL;
+
+	VERSION_FEATURE(Fearless)
+};
+
+template <class P>
+struct v6_1 : v6_0<P> {};
+
+template <class P>
+using latest_version = v6_1<P>;
```

Comment:
This file could probably do with some guidance for future authors of protocol-versioned features. Actually it makes a lot of sense after some squinting - the main thing I had trouble with was the meaning of type `P`.

Reply:
I added some comments there